full-service/src/db/models.rs

```diff
@@ -37,23 +37,40 @@ pub const TXO_INPUT: &str = "input";
 pub const TXO_OUTPUT: &str = "output";
 pub const TXO_CHANGE: &str = "change";
 
+/// An Account entity.
+///
+/// Contains the account private keys, subaddress configuration, and ...
 #[derive(Clone, Serialize, Identifiable, Queryable, PartialEq, Debug)]
 #[primary_key(id)]
 pub struct Account {
+    /// Primary key
     pub id: i32,
+    /// ???
```

Comment:
Is there a short explanation about why an account has both an 'id' (the DB table primary key) and an 'account_id_hex' (which appears to be the hex-encoded digest of the Account object)?

Reply:
The short explanation is wanting to do operations based on a recoverable identifier from the account_key (the account_id_hex is hex-encoded hash of the account's public key), as opposed to doing lookups based on the account_key bytes blob (this would be in a context where we didn't have the primary key). It may be worth reviewing whether we have many contexts where you need to do that lookup...

Reply:
Cool. My suspicion is that we can use uniqueness constraints somewhere (e.g. on the private key bytes) to make sure each account is never added to the table twice. In that case, an auto-incrementing primary key will probably work fine.